File: src/067. Maximum path sum II/067.rb
file_content = IO.read('p067_triangle.txt')

triangle = []
file_content.lines do |line|
  triangle << line.split.map(&:to_i)
end

1.upto(triangle.size - 1) do |row|
  triangle[row - 1] << 0
  0.upto(triangle[row].size - 1) do |col|
    triangle[row][col] += [triangle[row - 1][col], triangle[row - 1][col - 1]].max
  end
end

puts triangle.last.max

File: src/067. Maximum path sum II/067.py
triangle = []
with open('p067_triangle.txt') as f:
    for line in f:
        triangle.append([int(n) for n in line.split()])

for row in range(1, len(triangle)):
    triangle[row - 1].append(0)
    for col in range(0, len(triangle[row])):
        triangle[row][col] += max(triangle[row - 1][col], triangle[row - 1][col - 1])

print(max(triangle[-1]))
